motionplan/motionPlanner.go

```diff
@@ -43,9 +46,20 @@ func PlanMotion(ctx context.Context,
 	seedMap map[string][]frame.Input,
 	fs frame.FrameSystem,
 	worldState *frame.WorldState,
+	constraintSpec *pb.Constraints,
 	planningOpts map[string]interface{},
 ) ([]map[string][]frame.Input, error) {
-	return motionPlanInternal(ctx, logger, []*frame.PoseInFrame{dst}, f, seedMap, fs, worldState, []map[string]interface{}{planningOpts})
+	return motionPlanInternal(
+		ctx,
+		logger,
+		dst,
+		f,
+		seedMap,
+		fs,
+		worldState,
+		constraintSpec,
```

Comment:
Why not put the constraintSpec in the plannerOpts?

Reply:
Since planner opts is a `map[string]interface{}` here, there's always a chance that the user could pass in a struct that has that pre-filled.

I can always derive an unused key for the map, "if `internal1` is in use, try `internal2`, and so on" but that feels weird and hacky

Reply:
I think it just feels weird to me to have something like "planner options" be a catchall for the options we want to use except some of them (like constraint specifications).  Will this get fixed when we move away from constraints specified in the options?

Would it be less hacky to have a default name that is specified the way we do with the defaults?

Reply:
The `plannerOptions` struct we have behaves exactly as you describe. This could probably be renamed to `extraParams`, since that's what it is. I think that would solve the issue?

Reply:
I don't need it to be renamed.  I was wondering if we could cut down on the variables by shoving this into the same struct, but I guess the difference that I was missing before is that these are specified through arguments and could therefore conflict with the other location they could be specified.  Maybe this is fine then. 

I'm still looking for a response to the `ConstraintFromProto` idea I have in the below comment though

Reply:
That would reduce readability a good bit and add a bunch of bonus parsing and other complexity to go wrong. These are all internal private functions so extra params aren't a huge problem. Same reason why we don't put worldstate, and the framesystem, and the input map, also into the map[string]interface. :)

Replied to the other comment.